src/Covenant/Ledger.hs

```diff
@@ -0,0 +1,103 @@
+module Covenant.Ledger
+  ( LedgerAccessor (..),
+    LedgerDestructor (..),
+    LedgerUnwrapAccessor (..),
+    LedgerFieldAccessor (..),
+  )
+where
+
+-- | @since 1.0.0
+data LedgerAccessor
+  = LedgerUnwrap LedgerUnwrapAccessor
+  | LedgerField LedgerFieldAccessor
+  deriving stock
+    ( -- | @since 1.0.0
+      Eq,
+      -- | @since 1.0.0
+      Ord,
+      -- | @since 1.0.0
+      Show
+    )
+
+-- | An accessor for a type with only a single field and no variants.
+--
+-- @since 1.0.0
+data LedgerUnwrapAccessor
+  = UnwrapColdCommitteeCredential
+  | UnwrapHotCommitteeCredential
+  | UnwrapDRepCredential
+  | UnwrapConstitution
+  | UnwrapChangedParameters
+  deriving stock
+    ( -- | @since 1.0.0
+      Eq,
+      -- | @since 1.0.0
+      Ord,
+      -- | @since 1.0.0
+      Show
+    )
+
+-- | An accessor for a single-variant type with labelled fields, with a
+-- different accessor per field.
+--
+-- @since 1.0.0
+data LedgerFieldAccessor
```

Comment:
Is it really necessary to do it this way? That is: We _have_ to have some representation of the fields somewhere else, don't we? If so, couldn't we just have an AST node similar to the PureScript [Accessor](https://github.com/mlabs-haskell/purus/blob/5fb1996f98e68028369798e4b2a7820efe9dde3d/src/Language/PureScript/CoreFn/Expr.hs#L33C7-L33C43)? 

I didn't quite understand the rationale for restricting field accessors to the ledger types. If we have datatype declarations and all the types are annotated then couldn't we just have field accessors like this for _everything_? We'll have access to all of the information we need to deduce the return type & so on. Actually it should be simpler than PureScript b/c we don't have to worry about open/polymorphic rows.

Reply:
...not that we'd have rows at all, but inside the compiler the difference between a row-based record and the generic representation of a record is kind of stylistic.